### ISSUE TITLE
Cursor moment keys which are not passed through to the shell are confusing

### ISSUE THREAD

**Issue by halloleo** [akermu/emacs-libvterm#419]
In the vterm buffer _some_ cursor movement keys like `<left>` or `<up>` are passed through to the shell (via the `vterm-mode-map`). So far so good,

However many others (e.g. `C-<up>` or `C-x C-x`) move visually the cursor in the buffer but do not tell so the shell, because they are "Emacs cursor-moving commands" . This is pretty confusing. 

I think the best solution would be to allow the commands underlying these keys only in `vterm-copy-mode`. But how to find all Emacs cursor-moving commands (at least the built-in ones)?

**Comment by Sbozzolo** [comment]
I think that it is fine to have some keys that are not sent to the shell, actually, I prefer it in some cases. 
Consider `M-v`, as far as I know, it is not used by most shells or programs. Hence, I use that key to move around the buffer _without_ having to be in `vterm-copy-mode`. When I do so, I can immediately jump to the command line pressing any key. This is something useful if you just want to go back and look at the output of a command.

Also, the more keys we send to the shell, the "less Emacs vterm will feel", it would be just like running xterm in a buffer. This is not what we want: we want as much integration as possible.

**Comment by halloleo** [comment]
Get your point, @Sbozzolo . I still think at least all cursor movement commands _which stay in the current line_ should be mapped: It is very confusing to move the cursor with, say, `backward-word` and then typing a character which is inserted in the middle of the word where the terminal _thinks_ the cursor is...

**Comment by Sbozzolo** [comment]
I agree with that. We can collect the missing bindings in this thread, and I will add them.

**Comment by EnigmaCurry** [comment]
I've started by defining these:

```
  (define-key vterm-mode-map [C-left] 'vterm-send-M-b)
  (define-key vterm-mode-map [M-left] 'vterm-send-M-b)
  (define-key vterm-mode-map [C-right] 'vterm-send-M-f)
  (define-key vterm-mode-map [M-right] 'vterm-send-M-f)
  (define-key vterm-mode-map [C-up] 'vterm-send-C-p)
  (define-key vterm-mode-map [C-backspace] 'vterm-send-backspace)
  (define-key vterm-mode-map [C-up] 'vterm-send-up)
  (define-key vterm-mode-map [C-down] 'vterm-send-down)
```

**Comment by halloleo** [comment]
And:

key | cmd
-|-
M-right | `vterm-send-C-e`
M-left | `vterm-send-C-a`
A-right | `vterm-send-C-e`
A-left | `vterm-send-C-a`
C-home | `vterm--self-insert`
C-end | `vterm--self-insert`
